From faa975dd74df535960bf8f82990f352d022a12a5 Mon Sep 17 00:00:00 2001
From: Greg Hudson <ghudson@mit.edu>
Date: Fri, 12 Feb 2021 15:11:25 -0500
Subject: [PATCH] Restore krb5_set_default_tgs_ktypes()

Samba only uses the correct name (krb5_set_default_tgs_enctypes) if it
cannot find the old one in the library, so removing the name causes a
linker error for existing builds.

(cherry picked from commit 17ee97788611f8f8f4a6bd69968a9499f4db2215)

ticket: 8985
version_fixed: 1.19.1

(cherry picked from commit 3e36b25712d940a8e325abc407143634365b51d0)
---
 src/lib/krb5/krb/init_ctx.c  | 11 +++++++++++
 src/lib/krb5/libkrb5.exports |  1 +
 2 files changed, 12 insertions(+)

diff --git a/src/lib/krb5/krb/init_ctx.c b/src/lib/krb5/krb/init_ctx.c
index d2b70acad..bfa99d9eb 100644
--- a/src/lib/krb5/krb/init_ctx.c
+++ b/src/lib/krb5/krb/init_ctx.c
@@ -375,6 +375,17 @@ krb5_set_default_tgs_enctypes(krb5_context context, const krb5_enctype *etypes)
     return 0;
 }
 
+/* Old name for above function.  This is not a public API, but Samba (as of
+ * 2021-02-12) uses this name if it finds it in the library. */
+krb5_error_code
+krb5_set_default_tgs_ktypes(krb5_context context, const krb5_enctype *etypes);
+
+krb5_error_code
+krb5_set_default_tgs_ktypes(krb5_context context, const krb5_enctype *etypes)
+{
+    return krb5_set_default_tgs_enctypes(context, etypes);
+}
+
 /*
  * Add etype to, or remove etype from, the zero-terminated list *list_ptr,
  * reallocating if the list size changes.  Filter out weak enctypes if
diff --git a/src/lib/krb5/libkrb5.exports b/src/lib/krb5/libkrb5.exports
index 25141dfc5..df6e2ffbe 100644
--- a/src/lib/krb5/libkrb5.exports
+++ b/src/lib/krb5/libkrb5.exports
@@ -567,6 +567,7 @@ krb5_set_config_files
 krb5_set_debugging_time
 krb5_set_default_realm
 krb5_set_default_tgs_enctypes
+krb5_set_default_tgs_ktypes
 krb5_set_error_message
 krb5_set_password
 krb5_set_password_using_ccache
